pkg/master/master.go

```diff
@@ -570,16 +570,30 @@ func (m *Master) init(c *Config) {
 	requestInfoResolver := &apiserver.APIRequestInfoResolver{APIPrefixes: sets.NewString(strings.TrimPrefix(defaultVersion.Root, "/")), RestMapper: defaultVersion.Mapper}
 	apiserver.InstallServiceErrorHandler(m.handlerContainer, requestInfoResolver, apiVersions)
 
+	// allAPIVersions records all AVIVersions that are supported at /apis, in the
+	// form of "group/version"
+	allAPIVersions := []string{}
+
 	if m.exp {
 		expVersion := m.experimental(c)
 		if err := expVersion.InstallREST(m.handlerContainer); err != nil {
 			glog.Fatalf("Unable to setup experimental api: %v", err)
 		}
-		apiserver.AddApiWebService(m.handlerContainer, c.ExpAPIPrefix, []string{expVersion.Version})
+		g, err := latest.Group("experimental")
```

Comment:
We'll want to do that from Openshift, so we can add that when we get to
it.  Or you can add a field to MasterConfig now that is a list of external
groups.

On Sep 15, 2015, at 6:04 PM, Daniel Smith <notifications@github.com> wrote:

In pkg/master/master.go
<https://github.com/kubernetes/kubernetes/pull/13955#discussion_r39571743>:

>  	if m.exp {
>  		expVersion := m.experimental(c)
>  		if err := expVersion.InstallREST(m.handlerContainer); err != nil {
>  			glog.Fatalf("Unable to setup experimental api: %v", err)
>  		}
> -		apiserver.AddApiWebService(m.handlerContainer, c.ExpAPIPrefix, []string{expVersion.Version})
> +		g, err := latest.Group("experimental")

I'd like to see a loop here that goes through every Group, instead of
specifically referencing experimental. But we can fix this when we add
additional groups, I guess.

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/13955/files#r39571743>.